code_map.py

```diff
@@ -610,11 +696,37 @@ def run(self, edit):
         self.view.sel().clear()
         self.view.sel().add(line_region)
         sublime.set_timeout_async(
-            lambda: self.view.sel().add(line_region), 100)
+            lambda: self.view.sel().add(line_region), 10)
+
+# =============================================================================
+
+class code_map_temp_view_msg(sublime_plugin.TextCommand):
+
+    def run(self, edit):
+
+        map_view = get_code_map_view()
+        msg = "This view is not bound\n" + \
+              "to a file. Press\n\n    alt+m, alt+v\n\nto try to map it."
+        map_view.set_read_only(False)
+        map_view.set_scratch(False)
+        md = "Packages/Markdown/Markdown.sublime-syntax"
+        map_view.settings().set('syntax', md)
+        all_text = sublime.Region(0, map_view.size())
+        map_view.replace(edit, all_text, msg)
+        map_view.set_scratch(True)
+        map_view.set_read_only(True)
 
 # =============================================================================
 
 
+class code_map_ensure_group_closed(sublime_plugin.WindowCommand):
+
+    def run(self):
+        groups = self.window.num_groups()-1
+        if not self.window.views_in_group(groups):
+            reset_layout(reduce=True)
+
+
```

Comment:
Instead of restoring the layout from the listener, it calls this command with a timeout. I had crashes while switching projects with the old behaviour. Now it seems to cover all situations. Tested in a clean portable installation and in my main installation(that crashed with the previous listener)